cloud/aws/bin/dumpdb.py

```diff
@@ -0,0 +1,142 @@
+        print(f'Generating key pair in {tmpdir}')
+        _run_cmd(f'ssh-keygen -t rsa -b 4096 -f {tmpdir}/dbaccess -N ""')
+
+        print('Deploying dbaccess instance')
+        os.environ[
+            'TF_VAR_dbaccess_cidr_allowlist'] = f'["{_detect_public_ip()}/32"]'
+        os.environ['TF_VAR_dbaccess'] = "true"
+        os.environ['TF_VAR_dbaccess_public_key'] = f'{tmpdir}/dbaccess.pub'
+
+        try:
+            _run_terraform(config)
+
+            ec2_host_ip = aws.get_dbaccess_ec2_host_ip()
+            print(f'EC2 host IP is {ec2_host_ip}')
+
+            db_hostname = aws.get_database_hostname()
+            db_user = aws.get_secret_value(
+                config.app_prefix + '-civiform_postgres_username')
+            db_pwd = aws.get_secret_value(
+                config.app_prefix + '-civiform_postgres_password')
+
+            args = f'-o "UserKnownHostsFile=/dev/null" -o "StrictHostKeyChecking=no" -o "IdentitiesOnly=yes" -i {tmpdir}/dbaccess'
+            ssh = f'ssh -q {args} "ubuntu@{ec2_host_ip}" '
+
+            # We typically need about 15 seconds before the EC2 instance is ready
+            # to accept SSH connections.
+            print('Waiting for SSH access to EC2 host to become available')
+            sleep(15)
+            while True:
+                try:
+                    _run_cmd(ssh + 'exit', quiet=True)
+                    break
+                except subprocess.CalledProcessError as e:
+                    if e.returncode == 255:
+                        print(
+                            'SSH connection failed. Retrying in 10 seconds...')
+                        sleep(10)
+                    else:
+                        raise e
+
+            print('Installing postgresql-client')
+            cmd = ssh + '"sudo apt-get update && sudo apt-get install -y postgresql-client"'
+            _run_cmd(cmd)
+
+            print('Generating dump file')
+            cmd = ssh + f"PGPASSWORD='{db_pwd}' pg_dump -w -Fc -h '{db_hostname}' -U '{db_user}' -d postgres > civiform_database.dump"
+            _run_cmd(cmd)
+
+            print('Downloading dump file to local machine')
+            cmd = f'scp {args} "ubuntu@{ec2_host_ip}:/home/ubuntu/civiform_database.dump" {dumpfile}'
+            _run_cmd(cmd)
+
+            # Not strictly necessary, but in case the host sticks around for some reason.
+            print('Delete dump file on EC2 host')
+            cmd = ssh + 'rm -f civiform_database.dump'
+            _run_cmd(cmd)
+
+            input(
+                'Database dump complete. Press Enter to tear down the temporary resources.'
+            )
+        except:
+            input(
+                "\nError occurred. See details above. Press Enter to tear down the temporary resources."
+            )
+            raise
+        finally:
+            print('Cleaning up resources')
+            os.unsetenv("TF_VAR_dbaccess_cidr_allowlist")
+            os.unsetenv("TF_VAR_dbaccess")
```

Comment:
(question) Wouldn't these have been set in the python process and go out of scope when the app ends?